### DISCUSSION TITLE
get opinion through citation

### DISCUSSION THREAD

**Discussion by legaltextai** [freelawproject/courtlistener#4048] [Q&A] [answered]
Is there a way to get to an opinion text if I know the citation, like e.g. here the volume, reporter and page are in the url https://cite.case.law/minn/310/454/

**Comment by mlissner** [comment]
You'll need to look up the citation (and they're not unique), but once you do, you can get the text from that. Check out the Citation Lookup API: https://www.courtlistener.com/help/api/rest/v3/citation-lookup/

**Comment by legaltextai** [comment]
but it does not return the opinion_text or cluster_id or any other
connecting link to opinion_text. what am i missing?

**Reply by mlissner** [reply]
Can you share the query you made and the response?

**Comment by legaltextai** [comment]
I took the same examples as listed on the page

curl -X POST "https://www.courtlistener.com/api/rest/v3/citation-lookup/" \

  --header 'Authorization: Token my_token' \

  --data 'volume=576' \

  --data 'reporter=U.S.' \

  --data 'page=644'


so , to get to the text of the opinion , do one more step
extract cluster_id and run another API call with cluster_id?

**Reply by mlissner** [reply, marked as answer]
That's going to give you the cluster object, which has a key for sub_opinions. You'll find the text in the sub_opinion objects.

**Reply by legaltextai** [reply]
danke schon
